n_utils/includes/aws_domainjoin.sh

```diff
@@ -39,478 +39,472 @@ MAX_APPEND_DIGITS=5
-    for i in $(seq $SET_HOSTNAME_APPEND_NUM_DIGITS)
-    do
-        NUM_DIGITS=$(expr $NUM_DIGITS \* 10 )
-    done
-    NUM_DIGITS=$(expr $NUM_DIGITS - 1)
-
-    # If the number of digits is 5, $AWS_CLI_INSTALL_DIR/all_suffixes.txt will
-    # have numbers 00000-99999, one per line.
-    # If the number of digits is 2, the file will have numbers from 00 to 99, one per line.
-    for i in $(seq 0 $NUM_DIGITS)
-    do
-        if [ $SET_HOSTNAME_APPEND_NUM_DIGITS -eq 1 ]; then
-            printf "%01d\n" $i
-        fi
-        if [ $SET_HOSTNAME_APPEND_NUM_DIGITS -eq 2 ]; then
-            printf "%02d\n" $i
-        fi
-        if [ $SET_HOSTNAME_APPEND_NUM_DIGITS -eq 3 ]; then
-            printf "%03d\n" $i
-        fi
-        if [ $SET_HOSTNAME_APPEND_NUM_DIGITS -eq 4 ]; then
-            printf "%04d\n" $i
-        fi
-        if [ $SET_HOSTNAME_APPEND_NUM_DIGITS -eq 5 ]; then
-            printf "%05d\n" $i
-        fi
-    done > $AWS_CLI_INSTALL_DIR/all_suffixes.txt
-
-    MAX_RETRIES=5
-    for i in $(seq 1 $MAX_RETRIES)
-    do
-        get_list_of_computers > $AWS_CLI_INSTALL_DIR/ldap_search.txt
-        if [ ! -s $AWS_CLI_INSTALL_DIR/ldap_search.txt ]; then
-            echo "**Failed: Could not get list of computers"
-            exit 1
-        fi
-
-        # $AWS_CLI_INSTALL_DIR/ldap_filtered_hosts.txt has hosts
-        # with the same hostname prefix
-        # and then remove the hostname in each line so only the number suffixes remain.
-        # This gives us the list of suffixes that already exist.
-        cat $AWS_CLI_INSTALL_DIR/ldap_search.txt \
-            | grep -iP "^[ ]*$SET_HOSTNAME\d{$SET_HOSTNAME_APPEND_NUM_DIGITS}$" \
-            | sort | sed "s/$SET_HOSTNAME//gI" | tr -d ' ' \
-              > $AWS_CLI_INSTALL_DIR/ldap_filtered_hosts.txt
-
-        # The 'comm' utility does 'set complement' between
-        # $AWS_CLI_INSTALL_DIR/all_suffixes.txt
-        # and $AWS_CLI_INSTALL_DIR/ldap_filtered_hosts.txt
-        # to find suffixes that do not exist in ldap_filtered_hosts.
-        comm -23 $AWS_CLI_INSTALL_DIR/all_suffixes.txt \
-            $AWS_CLI_INSTALL_DIR/ldap_filtered_hosts.txt \
-                > $AWS_CLI_INSTALL_DIR/possible_suffixes.txt
-        if [ $? -ne 0 ]; then
-            echo "**Failed: Could not execute comm utility"
-            cleanup_temp_files
-            exit 1
-        fi
-
-        num_lines=$(wc -l $AWS_CLI_INSTALL_DIR/possible_suffixes.txt | awk '{ print $1 }')
-
-        if [ $num_lines -eq 0 ]; then
-           echo "**Failed: Could not find unused hostnames"
-           exit 1
-        fi
-
-        # When we have a list of possible hosts, one in each line, we can use
-        # a random number to choose one of the unused suffixes.
-        RANDOM_VALUE=$(head -c 512 /dev/urandom | xxd -p | tr -dc '0-9' \
-                            | fold -w $MAX_APPEND_DIGITS | head -n 1)
-        if [ -z $RANDOM_VALUE ]; then
-           LINE_N = 1
```

Comment:
Bug here, fixed it but might change script behaviour 🤔

Reply:
This script is not in use currently AFAIK. It's a work in progress